### PR TITLE
fix(deps): restore functionality of USE_EXISTING_SRC_DIR

### PR DESCRIPTION
**Pull request by jamessan**
30a0299bc removed the USE_EXISTING_SRC_DIR hack which broke building the
nightly PPA since ExternalProject tried to download the sources.

### CODE REVIEW

**Comment by dundargoc** [comment]
I think 30a0299bc6d1ceedd04d897cf56b298dd3ded0cd is the offending commit.

Do you mind adding a job in `.github/workflow/build.yml` that emulates the PPA build to prevent this in the future?

**Comment by jamessan** [comment]
> Do you mind adding a job in `.github/workflow/build.yml` that emulates the PPA build to prevent this in the future?

Do you want a completely separate job or add a couple steps to the old cmake job?

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -78,3 +78,30 @@ jobs:
...
+        run: make deps
+
+      - name: Clean bundled dependencies à la neovim/deps
+        run: |
+          rm -rf ./build
+          find .deps .deps/build -maxdepth 1 '!' \( -name .deps -o -name build -o -name src \) -exec rm -r '{}' +
+          cd .deps/build/src
+          rm -rf ./*-build
+          rm -rf ./*-stamp/*-{configure,build,install,done}
+          for d in *; do (cd "$d"; rm -rf ./autom4te.cache; make clean || true; make distclean || true); done
```
Comment (dundargoc) [inline .github/workflows/build.yml:101]: nitpick, not a blocker: is this necessary? The only files that should remain are the source files after the externalproject `PATCH_COMMAND`, so nothing should've been built I feel like. Not important though.

**Review by dundargoc — APPROVED** [review]
Looks good. Tested removing the `USE_EXISTING_SRC_DIR` and it fails as expected. Thank you.

**Pull request merged**